### ISSUE TITLE
EAST detect some texture as a text!

### ISSUE THREAD

**Issue by hahmad2008** [argman/EAST#358]
I am running the demo using the pre-trained model

`python eval.py --test_data_path=/tmp/images/ --gpu_list=0 --checkpoint_path=/tmp/east_icdar2015_resnet_v1_50_rbox/ \
--output_dir=/tmp/`

However, EAST detect some features as text, is there any reason behind that? Is there any configuration to tune or change?

Samples:

<img width="419" alt="Screen Shot 2020-08-24 at 9 41 49 PM" src="https://user-images.githubusercontent.com/68266028/91155986-3ed3b480-e6cc-11ea-932c-9d6c73d16a59.png">
<img width="142" alt="Screen Shot 2020-08-24 at 10 53 06 PM" src="https://user-images.githubusercontent.com/68266028/91156009-472bef80-e6cc-11ea-8dc7-57abae542b19.png">
<img width="111" alt="Screen Shot 2020-08-24 at 9 46 43 PM" src="https://user-images.githubusercontent.com/68266028/91156052-527f1b00-e6cc-11ea-8b47-7f89bbc0de0b.png">

**Comment by CarMiranda** [comment]
No, there is no "reason". EAST is just another deep learning model, so, as of right now, we cannot explain its outputs. And, again, as any other deep learning model, it can misbehave.

There are multiple things you can do:
- To improve accuracy in your particular data, you can tune the threshold parameters; maybe they are too low? This is not truly a recommendation since it does not improve your model...
- To improve overall robustness, train on more data, preferably adversarial to your case. Here, it predicts texture as text, so you can train on images with text and texture to force it to learn these kind of texture are not text. This is also known as hard example mining.
- To understand what is going on, try to interpret intermediate results (the score map `f_score` and geometry map `f_geometry`). Are intermediate results similar between these textures and text? If so, then the model has not learned to discriminate between them, and needs more training.

**Comment by francotheengineer** [comment]
@CarMiranda  answer is correct. 

You could also try using larger Backbones, more UNet layers, deeper box/score maps heads. 
Other mods could be changing activation functions, different Batch Norm layers. The list is endless.
